includes/polling/bgp-peers.inc.php

```diff
@@ -10,6 +10,9 @@
     if (!empty($peers)) {
         if ($device['os'] == 'junos') {
             $peer_data_check = snmpwalk_cache_long_oid($device, 'jnxBgpM2PeerIndex', '.1.3.6.1.4.1.2636.5.1.1.2.1.1.1.14', $peer_data_tmp, 'BGP4-V2-MIB-JUNIPER', 'junos');
+        } elseif ($device['os'] === 'dnos') {
+            $peer_data_checki = snmpwalk_group($device, 'bgpPeerEntry', 'BGP4-MIB');
+            $peer_data_check = array();
```

Comment:
You probably have a typo here. $peer_data_checki is wrong and gives you an empty array. (and you anyway empty the array after).

Reply:
It was intentional. The way it seemed to work was to poll each device separately if the peer_data_check variable had data. Renamed variable and created a string function with different snmp flag.

Reply:
Yep, $peer_data_check is here to know if there is at least 1 peer. If yes, we can continue and collect the list of them, parse it, and store in $peer_data what is needed. 

If you need/want to skip the check part (because you have no way to get the number of peers without getting the complete list of them for instance), then you could directly fill $peer_data  instead of creating $peer_data_checki. 

That being said, BGP4-MIB is the standard MIB so this data is already parsed, you should not have to rewrite it here, all the code is already included. If something is broken in it, it should probably be identified and fixed, instead of adding code duplication. 

What's wrong with current code ?

Reply:
Check the beginning of bgpwalk.txt I attached in the comment below, the current code implementation does not get any proper response from Dell since it queries specific peers and not the whole table, which does not seem to be supported by Dell. Unless this is some configuration error on my end?

So what I did was create new array for polling the whole table without specific peers and then in the peer loop part use the new array instead of polling each separately. But probably would be better to start using dellNetBgpM2 but that needs some more work.